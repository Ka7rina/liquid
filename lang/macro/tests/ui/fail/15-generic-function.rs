use liquid_lang as liquid;

#[liquid::contract(version = "0.1.0")]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    impl Noop {
        #[liquid(constructor)]
        fn init(&mut self) {}

        #[liquid(external)]
        fn noop<T>(&self, value: T) {}
    }
}

fn main() {}
