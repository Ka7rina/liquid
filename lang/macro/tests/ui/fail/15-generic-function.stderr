error: generic is not supported for functions in liquid contract
  --> $DIR/15-generic-function.rs:13:16
   |
13 |         fn noop<T>(&self, value: T) {}
   |                ^^^
